mtags/src/main/scala-2/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -131,57 +150,113 @@ object MultilineStringFormattingProvider {
+    case (token: Token.Constant.String, _)
+        if inToken(startPosition, endPosition, token)
+          && (if (andMultilineString) isMultilineString(token) else true)
+          && (if (andHadStripMargin) hasStripMarginSuffix(token) else true)
+          && (if (andHasPipeInScope)
+                pipeInScope(
+                  startPosition,
+                  endPosition,
+                  sourceText,
+                  newlineAdded
+                )
+              else true) =>
+      indent
+    case (token: Interpolation.Start, index: Int)
+        if token.pos.start < startPosition.start && {
+          var endIndex = index + 1
+          while (!tokens(endIndex)
+              .isInstanceOf[Interpolation.End]) endIndex += 1
+          val endToken = tokens(endIndex)
+          endToken.pos.end > endPosition.end &&
+          (if (andMultilineString) isMultilineString(token) else true) &&
+          (if (andHadStripMargin) hasStripMarginSuffix(endToken) else true) &&
+          (if (andHasPipeInScope)
+             pipeInScope(
+               startPosition,
+               endPosition,
+               sourceText,
+               newlineAdded
+             )
+           else true)
+        } =>
+      indent
   }
-  private def shouldFormatInterpolationString(
-      sourceText: String,
-      tokens: Tokens,
+
+  private def indentTokensOnTypeFormatting(
       start: StartPosition,
-      endPosition: EndPosition
-  ): PartialFunction[(Token, Int), Boolean] = {
-    case (token: Interpolation.Start, index: Int)
-        if token.pos.start <= start.start => {
-      var endIndex = index + 1
-      while (!tokens(endIndex)
-          .isInstanceOf[Interpolation.End]) endIndex += 1
-      isMultilineString(sourceText, token) && hasStripMarginSuffix(
-        endIndex,
-        tokens
-      ) && tokens(endIndex).pos.end > endPosition.end
-    }
+      end: EndPosition,
+      tokens: Tokens,
+      newlineAdded: Boolean,
+      splitLines: Array[String],
+      position: Position,
+      text: String
+  ): List[TextEdit] = {
+    implicit val sourceText: String = text
+    implicit val tokensImplicit: Tokens = tokens
+    implicit val newlineAddedImplicit: Boolean = newlineAdded
+    tokens.zipWithIndex
+      .collectFirst {
+        isStringOrInterpolation(
+          start,
+          end,
+          andMultilineString = true,
+          andHadStripMargin = true,
+          andHasPipeInScope = true
+        )(List(indent(splitLines, position))) orElse {
+          isStringOrInterpolation(
+            start,
+            end,
+            andMultilineString = true,
+            andHadStripMargin = false,
+            andHasPipeInScope = true
+          ) {
+            val newPos =
+              new Position(position.getLine, position.getCharacter + 3)
+            List(
+              indent(splitLines, position),
+              new TextEdit(new Range(newPos, newPos), ".stripMargin")
```

Comment:
I am not sure if we want to add `stripMargin` this automatically. It was previously done on purpose not to add it. The initial plan was to do it as in Intellij, but after some discussions it turns out it's not something that every developer wants and we shouldn't be forcing them.

Reply:
Ok, I didn't know that you changed your mind about the stripMargin. 
In my case, every time I add a pipe, it's because I will add stripMargin. It's quite annoying to write it, sometimes by error, the completion puts instead stringPrefix in VSCode, and it's quite slow for this case in Intellij. 
If the person doesn't want it, he will delete it. So it will depend on how many people do actually write the stripmargin compared to those who will delete it?

Reply:
It's a tough choice really. For me it would be useful, but I worry some people might be opposed.